### DISCUSSION TITLE
PB propensity for runaway memoryleaks? etc.

### DISCUSSION THREAD

**Discussion by pragmatech** [pocketbase/pocketbase#2633] [Q&A]
No ace developer here. However, I've been stoked to get PB running on my little vm (via caprover). Also, my new weirdo assistant Chat3PO has been getting me more to hitting my API with a nice little preact/vite frontend. Alas, my little server has once again sh*t its pants. Not a technical term I know, but nonetheless, I thought this a good chance to inquire of what kind of 'runaway' conditions (or is that not even a thing with PB?) I could be on the look out for when I now blithely swing react useEffect hooks to hit the api – coupled with not knowing half of what I'm doing! (good or bad, think 'Data', that orange Star Trek cat, as the main navigator at this point ...)
Thanks!

**Comment by ganigeorgiev** [comment]
What exactly do you mean by _"my little server has once again sh*t its pants."_?
Was the app terminated by some supervisor process due to high resources consumption or there was a Go panic? Do you keep any stdout/stderr logs?

In bursty/spiky scenarios the Go runtime may keep allocating memory faster than being able to "return" it to the OS and you may end up with OOM errors (_especially if you are memory constrained or there are multiple applications running on the server_). In this case you can try to set the env variable [`GOMEMLIMIT=500MiB`](https://pkg.go.dev/runtime@master#hdr-Environment_Variables) (_it is a soft-limit and will force the GC to run more often; adjust the value based on your available memory and the other active running processes_).

**Reply by pragmatech** [reply]
thanks for the reply (and this supercool tool). sorry for my bad terminology. I meant to actually convey that I woke up to my entire Google vm running PB on Caprover (Docker on Ubuntu more or less) had diminished to the nginx (502). It's the smallest 1 GB server, and it does break pretty readily when pushed at all. However, as it seemed to crash overnight, whereby usually I'm trying to install something that needs more... horsepower; it seemed an opportunity to learn if my fairly indiscriminate throwing of collections onto PB can cause such a thing. 
To that end, your reply has helped in my education regarding your little beast Thanks again!
ps- I'll try to get some of the log data on here upon getting the vm ... into some fresh diapers
